### PR TITLE
✨Implement AMP AsyncInput Base Class for amp-recaptcha-input

### PR DESCRIPTION
**Pull request by torch2424**
relates to #2273
relates to #18829

This implements the AMP.AsyncInput Class that extends BaseElement.

This will be used by `amp-recaptcha-input` and other upcoming features like [amp-google-pay-*](https://github.com/ampproject/amphtml/issues/15201) to allow for Asynchronous Inputs in Forms.

Please see the design doc in #2273 for more context

### CODE REVIEW

File: src/async-input.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    element.classList.add('i-amphtml-async-input');
+  }
+
+  /**
+   * Called to get the asynchronous value of an
+   * AsyncInput field.
+   * @return {!Promise<string>}
+   */
+  getValue() {
+    return Promise.resolve();
```
Comment (lannka) [inline src/async-input.js]: should it throw to force an override?

Comment (torch2424) [reply src/async-input.js]: Yes it should! Didn't know I could do that, I'll circle back with someone, make the change, and then merge this in, thank you! 😄

File: src/async-input.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  /**
+   * Called to get the asynchronous value of an
+   * AsyncInput field.
+   * @return {!Promise<string>}
+   */
+  getValue() {
+    let overrideError = new Error('Elements that implement AsyncInput' + 
+      ' must override getValue()');
+    throw new Error();
```
Comment (lannka) [inline src/async-input.js]: wrong error thrown :-)

**Comment by torch2424** [comment]
@lannka Made your changes, I know you already approved, but I just want you to see this that way you know I'm not committing that last commit haha!

**Pull request merged**